### ISSUE TITLE
Nice to have in the DSL

### ISSUE THREAD

**Issue by hsanjuan** [ipfs/kubernetes-ipfs#16]
I'm going to keep track here of a few things that cross my mind and that would be nice to have:

* Sucesses should be implicit and it should be expected that everything succeeds automatically
* Should stop testing if a test fails
* Should provide a way to keep an env per Pod so that when running in parallel in a bunch of nodes save_to SOMETHING in a pod does not overwrite the env for everyone else
* Should allow to run everywhere (i.e. on_node: -1)
* Should include networking operations support (i.e. cut network from pod X)
* Should include custom post-exec commands when auto-scaling

**Comment by dgrisham** [comment]
I haven't looked into the feasibility of this, but it'd be nice to do something like:

```yml
inputs:
   - HASH_1
   - HASH_2
   [...]
   - HASH_10
cmd: "for i in {1..10}; do
       ipfs cat ${HASH_$i}
      done"
```

But as it stands, I think it tries to input all 'global' variables (`$HASH_1`, etc.) before evaluating the 'local' ones (the iterator `i` in this case).  Not sure if this would be doable somehow or too much of a pain, but using a loop here can shorten the code quite a bit.

And on a similar note, it'd be nice to be able to do something to the effect of

```yml
input:
    - HASH_[1..10]
```

instead of what I have above.

**Comment by ZenGround0** [comment]
It would be nice to account for basic arithmetic when making substitutions during the parameterization pre-processing step.  For example the current cluster-test runner script needs to write 13 parameters to the config in order to substitute in all the necessary parameters.  However all 13 parameters are derivable from two root parameters by addition, multiplication and division by constants (e.g. `N: 4, Y: 3, N_times_4: 16, ParamA: 2*N*Y + Y/3...`).  Only including addition, multiplication and parentheses ordering would make parameterizing a lot cleaner.
